File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  location: string;
  area_sqft: number;
  floor_num: number;
  bathroom: number;
  balcony: number;
  parking: number;
  furnishing: string;
  transaction: string;
  ownership: string;
  facing: string;
  status: string;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel

class PredictionRequest(BaseModel):
    location: str
    area_sqft: float
    floor_num: int
    bathroom: int
    balcony: int
    parking: int
    furnishing: str
    transaction: str
    ownership: str
    facing: str
    status: str

class PredictionResponse(BaseModel):
    predicted_price: float
